specification.markdown

```diff
@@ -173,21 +173,55 @@ An EBML Stream is a file that consists of one or many EBML Documents that are co
+
+Within the EBML Schema each EBML Element is defined to occur at a specific level. For any specificied EBML Element that is not at level 0, the Parent EBML Element refers to the EBML Master-element that that EBML Element is contained within. For any specifiied EBML Master-element the Child EBML Element refers to the EBML Elements that may be immediately contained within that Master-element. For any EBML Element that is not defined at level 0, the Parent EBML Element may be identified by the preceding `<element>` node which has a lower value as the defined `level` attribute. The only exception for this rule are Global EBML Elements which may occur within any Parent EBML Element within the restriction of the Global EBML Element's range declaration.
+
+#### EBML Schema Element Attributes
+
+Within an EBML Schema the `<element>` uses the following attributes to define an EBML Element.
+
+| attribute name | required | definition |
+|----------------|----------|------------|
+| name           | Yes      | The official human-readable name of the EBML Element. The value of the name MUST be in the form of an NCName as defined by the [XML Schema specification](http://www.w3.org/TR/1999/REC-xml-names-19990114/#ns-decl). |
+| level          | Yes      | The level notes at what hierarchical depth the EBML Element may occur within an EBML Document. The initial EBML Element of an EBML Document is at level 0 and the Elements that it may contain are at level 1. The level MUST be expressed as an integer; however, the integer may be followed by a '+' symbol to indicate that the EBML Element is allow valid at any higher level.  |
+| global         | No       | A boolean to express if an EBML Element MUST occur at its defined level be may occur within any Parent EBML Element. |
+| id             | Yes      | The Element ID expressed in hexadecimal notation prefixed by a '0x'. |
+| mandatory      | No       | A boolean to express if the EBML Element MUST occur if the Parent EBML Element is used. If the mandatory attribute is not expressed for that Element then that element is to be considered not mandatory. |
+| multiple       | No       | A boolean to express if the EBML Element may occur within its Parent EBML Element more than once. If the multiple attribute is false or the  multiple attribute is not used to define the Element then that EBML Element MUST not occur more than once with that Element's Parent EBML Element. |
+| range          | No       | For Elements which are of numerical types (Unsigned Integer, Signed Integer, Float, and Date) a numerical range may be specified. If specified that the value of the EBML Element MUST be within the defined range inclusively. |
```

Comment:
@mjbshaw, the EBML Schema concept of `range` is analogous to the XML Schema concept of `restriction`, so I propose aligning with the [specification of the `restriction` expression](http://www.w3schools.com/Xml/schema_facets.asp) to the extent feasible.

To document what currently exists (only the short list above), I suggest we could say:

- a prefix of the string `not ` negates the value of `range`
- the `range` must be expressed in of the these forms:
    - `x-y` where x and y are integers and y must be greater than x
    - `>x` where x is an integer, meaning that the value must be greater than `x`
    - `x` where x is an integer, meaning that the value must equal `x` (in the current schema this is only used with negation via `not `)
- white space may appear for readability but is not meaningful
- `range` are only defined for integer ranges
- `range` does not support date or float ranges

These rules serve to document what is currently there. Once that is in place, I propose we can proceed to change current practice towards the rules of the XML Schema `restriction`.